tools/unitctl/unit-client-rs/src/unitd_docker.rs

```diff
@@ -0,0 +1,403 @@
+                .or_else(|err| {
+                    Err(UnitClientError::UnitdDockerError {
+                        message: err.to_string(),
+                    })
+                });
+
+            let resp: ContainerCreateResponse;
+            match docker
+                .create_container::<String, String>(
+                    None,
+                    Config {
+                        image: Some(image.clone()),
+                        host_config: Some(HostConfig {
+                            network_mode: Some("host".to_string()),
+                            mounts: Some(mounts),
+                            ..Default::default()
+                        }),
+                        ..Default::default()
+                    },
+                )
+                .await
+            {
+                Err(err) => {
+                    return Err(UnitClientError::UnitdDockerError {
+                        message: err.to_string(),
+                    })
+                }
+                Ok(response) => resp = response,
+            }
+
+            let mut list_container_filters = HashMap::new();
+            list_container_filters.insert("id".to_string(), vec![resp.id]);
+            match docker
+                .list_containers::<String>(Some(ListContainersOptions {
+                    all: true,
+                    limit: None,
+                    size: false,
+                    filters: list_container_filters,
+                }))
+                .await
+            {
+                Err(e) => Err(UnitClientError::UnitdDockerError { message: e.to_string() }),
+                Ok(info) => {
+                    if info.len() < 1 {
+                        return Err(UnitClientError::UnitdDockerError {
+                            message: "couldnt find new container".to_string(),
+                        });
+                    }
+                    if info[0].names.is_none() || info[0].names.clone().unwrap().len() < 1 {
+                        return Err(UnitClientError::UnitdDockerError {
+                            message: "new container has no name".to_string(),
+                        });
+                    }
+
+                    match docker
+                        .start_container(
+                            info[0].names.clone().unwrap()[0].strip_prefix("/").unwrap(),
+                            None::<StartContainerOptions<String>>,
+                        )
+                        .await
+                    {
+                        Err(err) => Err(UnitClientError::UnitdDockerError {
+                            message: err.to_string(),
+                        }),
+                        Ok(_) => Ok(resp.warnings),
+                    }
+                }
+            }
+        }
+        Err(e) => Err(UnitClientError::UnitdDockerError { message: e.to_string() }),
+    }
+}
+
+/* Returns either 64 char docker container ID or None */
+pub fn pid_is_dockerized(pid: u64) -> bool {
```

Comment:
I'd encourage you to make this a bit more generic or to make a mechanism to fork logic between docker, podman, and lxc/lxd. At the least, it would be good to see how the CLI works with unit running in a lxd container.

Also, we may want to rethink using the name `docker` unless we specifically mean the Docker implementation.

Reply:
Here it specifically means dockerized (see the hardcoded docker in the regex) but I think this would be relatively easy to generalize.

Reply:
I think that we should roadmap additional container support and then move towards refactoring this into a more general module.